File: app/src/main/java/com/example/josycom/flowoverstack/viewmodel/CustomSearchViewModelFactory.java
package com.example.josycom.flowoverstack.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.josycom.flowoverstack.repository.SearchRepository;


public class CustomSearchViewModelFactory implements ViewModelProvider.Factory {

    private String inTitle;
    private SearchRepository searchRepository;

    public CustomSearchViewModelFactory(SearchRepository searchRepository) {
        this.searchRepository = searchRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new SearchViewModel(searchRepository);
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/ui/viewmodel/CustomSearchViewModelFactory.kt
package com.josycom.mayorjay.flowoverstack.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.josycom.mayorjay.flowoverstack.data.SearchRepository
import javax.inject.Inject

class CustomSearchViewModelFactory @Inject constructor(private val searchRepository: SearchRepository) : ViewModelProvider.Factory {

    private var page = 0
    private var pageSize = 0

    fun setInputs(page: Int, pageSize: Int) {
        this.page = page
        this.pageSize = pageSize
    }

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return SearchViewModel(page, pageSize, searchRepository) as T
    }
}